keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -247,11 +249,34 @@ class SqlArtifactRepository(
           .getValues(ARTIFACT_VERSIONS.VERSION)
       }
         .sortedWith(artifact.versioningStrategy.comparator)
+        .also { versions ->
+          return if (artifact is DockerArtifact) {
+            filterDockerVersions(artifact, versions)
+          } else {
+            versions
+          }
+        }
     } else {
       throw NoSuchArtifactException(artifact)
     }
   }
 
+  /**
+   * Given a docker artifact and a list of docker tags, filters out all tags that don't produce exactly one capture
+   * group with the provided regex.
+   *
+   * This means that this will filter out tags like "latest" from the list.
+   */
+  private fun filterDockerVersions(artifact: DockerArtifact, versions: List<String>): List<String> =
+    versions.filter { version ->
+      try {
+        version != "latest" && TagComparator.parseWithRegex(version, artifact.tagVersionStrategy, artifact.captureGroupRegex) != null
```

Comment:
I suggest pulling this line out into a function, since it's used twice (here and on line 762) and it's a complex enough concept that I think it's worth encapsulating into a function.